### PR TITLE
fix: flush logger

### PR DESCRIPTION
**Pull request by mkenigs**
Not flushing the logger appears to be causing flaky tests on macOS - likely the error messages expected by some tests are not being printed to stderr before flox exits.

Re-enable tests that were disabled due to this failure.

### CODE REVIEW

File: crates/flox/src/utils/init/logger.rs [hunk 1]
```diff
@@ -113,3 +115,15 @@ pub fn init_logger(verbosity: Option<Verbosity>, debug: Option<bool>) {
...
+pub fn flush_logger() {
+    if LOGGER_HANDLE.get().is_none() {
+        return;
+    }
+
+    let (_, fmt_handle) = LOGGER_HANDLE.get().unwrap();
+
+    let _ = fmt_handle.modify(|l| {
+        let _ = l.writer_mut().flush();
+    });
```
Comment (mkenigs) [inline crates/flox/src/utils/init/logger.rs:128]:
```suggestion
    match LOGGER_HANDLE.get() {
        None => return,
        Some((_, fmt_handle)) => {
            let _ = fmt_handle.modify(|l| {
                let _ = l.writer_mut().flush();
            });
        },
    }
```

**Comment by mkenigs** [comment]
Three green runs in a row 👀

**Review by zmitchell — APPROVED** [review]

**Comment by zmitchell** [comment]
Are you waiting on CI to see if you get another green test run or can we rebase and auto-merge?

**Comment by mkenigs** [comment]
Just got green run number 4 so I enabled auto merge

**Comment by mkenigs** [comment]
No user facing changes

**Pull request merged**